commit c06b301f478b2a0848e8453f5f71bf4f992e3193
Author: Jonathan M Hsieh <jmhsieh@apache.org>
Date:   Tue Dec 16 16:12:27 2014 -0800

    HBASE-12709 [mvn] Add unit test excludes command line flag to the build

diff --git pom.xml pom.xml
index 6c84c41..5c946b0 100644
--- pom.xml
+++ pom.xml
@@ -523,6 +523,12 @@
             <!--Allocate some direct memory for direct memory tests-->
             <argLine>-enableassertions -XX:MaxDirectMemorySize=1G -Xmx1900m -XX:MaxPermSize=256m -Djava.security.egd=file:/dev/./urandom -Djava.net.preferIPv4Stack=true -Djava.awt.headless=true</argLine>
             <redirectTestOutputToFile>${test.output.tofile}</redirectTestOutputToFile>
+            <excludes>
+	      <!-- users can add -D option to skip particular test classes
+		   ex: mvn test -Dtest.exclude.pattern=**/TestFoo.java,**/TestBar.java
+	      -->
+	      <exclude>${test.exclude.pattern}</exclude>
+	    </excludes>
           </configuration>
           <executions>
             <execution>
@@ -980,6 +986,7 @@
     <surefire.testFailureIgnore>false</surefire.testFailureIgnore>
     <test.output.tofile>true</test.output.tofile>
     <surefire.timeout>900</surefire.timeout>
+    <test.exclude.pattern></test.exclude.pattern>
   </properties>
   <!-- Sorted by groups of dependencies then groupId and artifactId -->
   <dependencyManagement>
